### ISSUE TITLE
🐛 iOS optimization level bug

### ISSUE THREAD

**Issue by github984527** [cuvent/react-native-vision-camera#561]
### What were you trying to do?

for ios,if the optimization level is faster ,the frame proccesser Plugins will  don't work 

### What happened instead?

optimization level is faster or high

### Device

iphone11 ios14.7

### VisionCamera Version

^2.8.0

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by mrousavy** [comment]
what is an optimization level?

**Comment by github984527** [comment]
it's an xocde build setting item。

Specifies the degree to which the generated code is optimized for speed and binary size.

None[-O0]: Do not optimize. With this setting, the compiler's goal is to reduce the cost of compilation and to make debugging produce the expected results. Statements are independent: if you stop the program with a breakpoint between statements, you can then assign a new value to any variable or change the program counter to any other statement in the function and get exactly the results you would expect from the source code.
Fast[-O1]: Optimizing compilation takes somewhat more time, and a lot more memory for a large function. With this setting, the compiler tries to reduce code size and execution time, without performing any optimizations that take a great deal of compilation time. In Apple's compiler, strict aliasing, block reordering, and inter-block scheduling are disabled by default when optimizing.
Faster[-O2]: The compiler performs nearly all supported optimizations that do not involve a space-speed tradeoff. With this setting, the compiler does not perform loop unrolling or function inlining, or register renaming. As compared to the 'Fast' setting, this setting increases both compilation time and the performance of the generated code.
Fastest[-O3]: Turns on all optimizations specified by the 'Faster' setting and also turns on function inlining and register renaming options. This setting may result in a larger binary.
Fastest, Smallest[-Os]: Optimize for size. This setting enables all 'Faster' optimizations that do not typically increase code size. It also performs further optimizations designed to reduce code size.
Fastest, Aggressive Optimizations[-Ofast]: This setting enables 'Fastest' but also enables aggressive optimizations that may break strict standards compliance but should work well on well-behaved code.

**Comment by mrousavy** [comment]
Okay and what does "don't work" mean? Do they not run? Do they crash? Are they slow? What gets printed to the Xcode logs/console? Need more details here.

**Comment by mrousavy** [comment]
Also, do Reanimated animations (UI Worklets) work with that setting?

**Comment by github984527** [comment]
On the iOS side, I wrote my own frame processor for facial recognition. Everything was normal during debugging, but the frame plugin did not run after the archive was packaged. At first I thought it was a code problem. After two days, I didn’t find any code. The question, after thinking about it again, I found that the main problem is that it doesn't work after packaging. I think there may be something wrong with the compilation and optimization. I tried to set the Optimization Level to none and re-packaged it to run. The frame plug-in was indeed running normally. So I think it may be that the optimization level affects the injection of the frame plug-in method. I am not sure about the specifics.
